util/start_test

```diff
@@ -784,6 +784,12 @@ def set_up_general():
         except:
             logger.write("[Error: Could not find valgrind.]")
             finish()
+
+        if "CHPL_REGEXP" in os.environ:
+            if "CHPL_RE2_VALGRIND_SUPPORT" not in os.environ:
+                logger.write("[Error: Rebuild re2 with CHPL_RE2_VALGRIND_SUPPORT set.]")
```

Comment:
I don't know how I feel about this check. I guess it's better than nothing, but since it's a build time option this check can be wrong: 
 - if you had `CHPL_RE2_VALGRIND_SUPPORT` set at build time, but not at start_test time, it will incorrectly error
 - if you have `CHPL_RE2_VALGRIND_SUPPORT` set at start_test time, but not build time, this gives you a false sense of security